daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -35,13 +35,22 @@ private[lf] object Speedy {
   private val enableLightweightStepTracing: Boolean = false
 
   /** Instrumentation counters. */
-  final case class Instrumentation(
-      var classifyCounts: Classify.Counts,
-      var countPushesKont: Int,
-      var countPushesEnv: Int,
-      var maxDepthKont: Int,
-      var maxDepthEnv: Int,
-  ) {
+  final case class Instrumentation() {
```

Comment:
I really dislike using case class for objects that contains mutable state.

Reply:
Good point. And in this case it would be trivial to make this a class. I'll get that done!